pilot/pkg/model/sidecar.go

```diff
@@ -68,6 +69,14 @@ type SidecarScope struct {
 	// Union of services imported across all egress listeners for use by CDS code.
 	services []*Service
 
+	// Set of all services this sidecar depends on.
+	// This field and similar fields below will be used to determine the config/resource scope
+	// which means which config changes will affect the proxies with this scope.
+	serviceDependencies map[host.Name]struct{}
```

Comment:
What is the difference between this and the services field above?

Reply:
the services field above is a slice and iterating it to check if a host exist is slow ...

Reply:
I agree these attributes should be merged